### DISCUSSION TITLE
Run as Non Root

### DISCUSSION THREAD

**Discussion by metalshanked** [prodrigestivill/docker-postgres-backup-local#108] [Q&A] [answered]
Is is possible to run the container as a Non Root user? Thanks!

**Comment by prodrigestivill** [comment, marked as answer]
Currently this project is using the same behaviour as the official postgresql server container so currently is running as root user.

Changing this would break any existing deployment, but this behaviour can be changed at runtime time by configuring it in your deployment configuration:

* If you use docker manually use the **--user** argument as described in [their documentation](https://docs.docker.com/engine/reference/run/#user).
* If you are using docker-compose set the **user** property at the docker-compose file as described in [their documentation](https://docs.docker.com/compose/compose-file/#domainname-hostname-ipc-mac_address-privileged-read_only-shm_size-stdin_open-tty-user-working_dir).
* If you are using Kubernetes use the **runAsUser** of the securityContext as described in [their documentation](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#podsecuritycontext-v1-core).

There are already created user and group in the images that can be used with the above configurations:
* For debian image (default) the user postgres (uid=999) and the group postgres (gid=999) can be used.
* For alpine image the user postgres (uid=70) and the group postgres (gid=70) can be used.
